system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs

```diff
@@ -61,6 +61,7 @@ impl Chain for BridgeHubKusama {
 
 impl Parachain for BridgeHubKusama {
 	const PARACHAIN_ID: u32 = BRIDGE_HUB_KUSAMA_PARACHAIN_ID;
+	const MAX_HEADER_SIZE: u32 = MAX_BRIDGE_HUB_HEADER_SIZE; // FAIL-CI @svyatonik please check
```

Comment:
I think this is good like this - @bkontur please confirm

```suggestion
	const MAX_HEADER_SIZE: u32 = MAX_BRIDGE_HUB_HEADER_SIZE;
```

Reply:
Yes, it is ok.  [MAX_BRIDGE_HUB_HEADER_SIZE](https://github.com/paritytech/polkadot-sdk/blob/master/bridges/chains/chain-bridge-hub-cumulus/src/lib.rs#L42-L43) is 4_096.

And I checked the last few imported BridgeHub headers, and the actual value is `228`.

```
Recently imported bridge-hub-kusama headers:
    3,534,269: 0x50e302c0a0ae09e3eaf08d524135ee409588f1cd0cacb7e9dea675eb39f51273 (228 bytes) - {"parentHash":"0xf8196641466d729035172b164d0fc3ccc623b81fde4635eddbd7f00717851161","number":3534269,"stateRoot":"0x9b2d538504453cef73d7d030c0f212e611bcbaf32a621568e0e24a8dbe4e7101","extrinsicsRoot":"0xf6b3b86549f6189cfcc357cdbad170a9568186765360f1b911831568482afdc4","digest":{"logs":[{"preRuntime":["0x61757261","0xb671890800000000"]},{"consensus":["0x52505352","0x25e0d2df96bcfaf541d38c6517f01d3fe53fae6f56b981157a7fdeb87fe02ca8fe62a405"]},{"seal":["0x61757261","0x3eb1982a17aa05edb3411d20c7e514e47ec183cfef814bc5cdcb158c07f9881b3bb9431008dfb7ba261f7322fe7a4638fc64b935d0522824149d3800b3289886"]}]}}
2024-07-11 11:19:34        REGISTRY: Unknown signed extensions BridgeRejectObsoleteHeadersAndMessages, RefundBridgeHubPolkadotMessages found, treating them as no-effect
    3,481,500: 0xc9397d878d9dd49e626fd57e4dee487bf33b1940a6ee7c4427df46686eebf9e9 (228 bytes) - {"parentHash":"0xeb0f81120fe3e051be79a095e36f3517c22b26181d286e003be55f9476ba463a","number":3481500,"stateRoot":"0x95470b84eaf440ef1975723adafc8edc9be6bb84d34c24bb05431e502f324248","extrinsicsRoot":"0x48b9740853ceb7496857856b919306bfedc3c0c1e8f015854e6d881364ba7ddb","digest":{"logs":[{"preRuntime":["0x61757261","0xbc96880800000000"]},{"consensus":["0x52505352","0x71545cb90a23b0003bdaf944aa2c4d00667c45ed42c77719b29432e63ec03c85c2939d05"]},{"seal":["0x61757261","0xb858b37ecf020c05b22b6da99197df1204ef8e07701f59f4dd1eabbd016285094aa6dab28c38ac53c9b1d4a1bc2e7a9d9ce5109752e88fa00af30340b720958b"]}]}}
    3,481,383: 0xabe7820a29b7c9327a042fca36c84dc12966b9785cadd54dadc5a8c746de6a88 (228 bytes) - {"parentHash":"0x0ec9a00a852f71df591303b31ea3124175835f4561da71694d560c0fc923acfd","number":3481383,"stateRoot":"0x74514d1fc6178bb247f8f56d88cce30e5a750be0df96d45c4d75d2bb0096a0dc","extrinsicsRoot":"0xd09b0914bdd95bbc259fcc6d430291d5b498e30d63c0e99c9b75f7f4e130d3f5","digest":{"logs":[{"preRuntime":["0x61757261","0x4096880800000000"]},{"consensus":["0x52505352","0xcebb0c891809509eb8a149aa3260169197dc84fcf76cb1123a6f277804e358b4de8f9d05"]},{"seal":["0x61757261","0x08f2f4f4ce3ba647eaeb810d61a680322e5cd044673cf943a2718bfde6a4374ed17835f444617f20b0161b59d8b0475f16063ad481051673d56c77a2b66c7887"]}]}}
    3,467,094: 0xb0cef85d234e071c7528e49290cc519743ca1aaa1f1140dbd8d2b01f3c4c8d85 (228 bytes) - {"parentHash":"0x07f00090474f02b693ed5dfd6fd759b5074b183f9c79f805cec4ff87f841a8b4","number":3467094,"stateRoot":"0xe6a894cd53f8891550976dfe902a4e8e892a7e23aacbbb5d95847b72b097fa05","extrinsicsRoot":"0xc0c547d29bd8297610f12a7836f62a1191ba1262fcae39bf4dead223bc47d011","digest":{"logs":[{"preRuntime":["0x61757261","0xea59880800000000"]},{"consensus":["0x52505352","0x3af657772e6e022090b76fc9e70a708f28b5a8b7ec4607ddcdfda8db68710a10deae9b05"]},{"seal":["0x61757261","0x9e3114e67b466fbfc2aa92a86bb6b8482666dfa933dd2156a3824ad43c1c8a32eda962b50a19be98822c3d750ed74715be12df8b59fa9fe028a8e99a9ab75f80"]}]}}
    3,448,668: 0x8e53299a3e33dabe8e9dfe772f6c7b1ab743613b9f2121c6ba17c38c73cfddba (228 bytes) - {"parentHash":"0xe12581dc89a5e6ef50e46c1bdaac9f962ca3f48c1fb0b98c49eb5746a84e004e","number":3448668,"stateRoot":"0x149e5604067762e1c7f17766d3751b41e4c3ce06351582420119ea78e91ad837","extrinsicsRoot":"0xa35fea17fba1278f21c455a4e1f75cff911f3f96d9ff3e01f6e6925be2d3606a","digest":{"logs":[{"preRuntime":["0x61757261","0xfb0b880800000000"]},{"consensus":["0x52505352","0xd2efd251b034375b8b44619ed4af79303d6318737e96c9384bd936eb6659083212429905"]},{"seal":["0x61757261","0x2a6ab59699e404fe4bfca064fb6ced2e2bb46ea4b98ae3f1ac52f87f4444e963f15c530117fb2dc796854fc615101d528e0b17706fa673b9d90d546bff7d8785"]}]}}
    3,448,616: 0x77619d84676153efada900a87ceba25e6e010d597c1b91f5c49beb58d8db2cc9 (228 bytes) - {"parentHash":"0x3a05a66b4af06435ff60823fe27fd8e6dace5aebbf6fdb51e813edc6cf44a723","number":3448616,"stateRoot":"0xc8d3366f4b8c65165b11f198fe4a0f7f11db4b5c4494a9627eed5122be00b482","extrinsicsRoot":"0x30a3a10dc81b0099f28112fa0100caef9de1637c0e75c1568a2d544cb2749371","digest":{"logs":[{"preRuntime":["0x61757261","0xc30b880800000000"]},{"consensus":["0x52505352","0x2177bd83fc1a20a0ba63c16c5ae05f3b3f71e5160dd5424fe55a1c5b20295d9256409905"]},{"seal":["0x61757261","0x5e4a54722c3144b416037059dcf63929ac203be6676cc75ba06d0158d3bb223f77b92e4651f4d904b627bfe0def8da22acfa53dd180c9e5d9a37b1a1f977418a"]}]}}
2024-07-11 11:19:38        REGISTRY: Unknown signed extensions BridgeRejectObsoleteHeadersAndMessages, RefundBridgeHubPolkadotMessages found, treating them as no-effect
    3,389,451: 0x44e7174b03b7019d08e1e71087adf24461ca622df2352aa7b4be6c5f20fec63b (228 bytes) - {"parentHash":"0x815945cabcd4b5a01c5df9bd20969a3ea76762586dc1dee06f0fee36befa13d7","number":3389451,"stateRoot":"0x4759cf9a910899f2e428b3bb6d97dd95a91e22fba8f88eec1db8974e7390f584","extrinsicsRoot":"0xfcd5f76439d835810b3c971f8dab2053048235a4a82d9cfcdf66ca3a93acb797","digest":{"logs":[{"preRuntime":["0x61757261","0xf310870800000000"]},{"consensus":["0x52505352","0xecc28f788839d2dc6b308f711db85de77e53b2c69a66fd58861078c4a1294873aa739105"]},{"seal":["0x61757261","0x2c7125ad882aa093ccfcea5b01ed11d155ced825f504ca48aa78584fa467022a58e6d0d4474dc6a1289ff0e6262884881375ccc88c14176a0bc4d8de9acd8887"]}]}}
    3,351,617: 0xf2c5360dda26f1eba1b94226268642854801ed2e1910e67ffd82686b1bde594e (228 bytes) - {"parentHash":"0xfb0e3677c1182b9436872d1158c538d6d9ff382d2432fa85562dbdc22a33426b","number":3351617,"stateRoot":"0x55eb6dc67fb968d100305a68fcda1802cc89da9fb9baf3c998bc06058eab31dd","extrinsicsRoot":"0x2628ffa266dd5b1b0b7e7a2d558ecd45ef8bdfacdf5ba9b24d6002781da28e40","digest":{"logs":[{"preRuntime":["0x61757261","0x0d70860800000000"]},{"consensus":["0x52505352","0xe2b5b8077263d3acd27c504497bc8d95a469e5b7d986f980cc5b53d0f109b8e2ae758c05"]},{"seal":["0x61757261","0x2cb0ffca125ed67c815911a0d15a393f7b5525a066d543b7e2be3fa5b2ab5534448de376566da11145610f1982219287a6d6479c628352c5aed1a515f4a1df85"]}]}}
    3,351,472: 0xff0c8226373ada32610e36831275fb8ba7963f48aacdbe573c46b415e4fab629 (228 bytes) - {"parentHash":"0x29d80d7e672d2f2fbf04f388a1ed76e1df9fd100cfef324b764533bd17b33192","number":3351472,"stateRoot":"0x0a6fe8027a4cfeb829f031de7d727643e5d441b3e79ba067ebb832c6ab1fbe30","extrinsicsRoot":"0x49d08685de089165283567345044742af79767135ede6dc454bdef55763a701e","digest":{"logs":[{"preRuntime":["0x61757261","0x766f860800000000"]},{"consensus":["0x52505352","0xbbb38e2e86ee974aab24b0db70abae0f3ab2016c944cf88bfa0eddb1462d22b4f6708c05"]},{"seal":["0x61757261","0x9a789e442bbf5bdc5168dcc08fa9a75b3759aaa3505347c6bbdb26f59a30ad42ccc8423df23b71998cceafc47f588d69fd0da61617a506946e51319026efbf87"]}]}}
    3,347,889: 0x9576902e035ee714826bb5dcb78fa563bb3127e551e2396ff445a6cf01882bc3 (228 bytes) - {"parentHash":"0x587f85dc043b505f43f0f66e9c06fc76b8ca128ae8e53775de036592856e037b","number":3347889,"stateRoot":"0x21b1967b82406a9dde0711326c6e8a28917c99f9f132525dbfb3eeeb84a04b26","extrinsicsRoot":"0xf305bc2d85fd13240e40f9ea81e52dbe5516729b5745c192fa4c9b9e77ce3651","digest":{"logs":[{"preRuntime":["0x61757261","0x3360860800000000"]},{"consensus":["0x52505352","0x37ef47165e5da6b755fb8b0f7e011e6fcaa5dcb9d310e6135934e2c9c857a94d32f78b05"]},{"seal":["0x61757261","0x584686b2193313e8187ac7cba2f59e9e4aa91bf1c57aaffbcb3b162f30fc5155082b12e84083f5f5c3440d09be61acb60c8a0cf9351ca8ed1154c62f209b998f"]}]}}
```

```
Recently imported bridge-hub-polkadot headers:
    2,819,481: 0xbceb3e1f2307911cd6da86ac7c2320d7431b731d33c6e4aa4f663d2ef66cd584 (228 bytes) - {"parentHash":"0x5e7f0560f11beca4c96f5eb772090f85bdda4a10fd50b9b56a94eb23cf7b9d60","number":2819481,"stateRoot":"0xa96a47e585d79b856d025f4d647e80692d7fd960146bc92fc24658fd4b136dad","extrinsicsRoot":"0x13bae72930ee2857b081724b1b9700918758fd15b1312ac9256eb5423fdc7e9b","digest":{"logs":[{"preRuntime":["0x61757261","0xb171890800000000"]},{"consensus":["0x52505352","0x120a6377a850bd321ae3ecb98ae66dfd2a1c10c6a06ec72e56676bbaddff7d53b2271205"]},{"seal":["0x61757261","0xecbbd4ffb40a9ae78374fa7514a78462a4d338e3517a174e11b6b8676f87a0232c5ca0f4d3be60973aa97dfbbdd79ff65de80e0a868194e874ff68dc2fb39b81"]}]}}
2024-07-11 11:19:46        REGISTRY: Unknown signed extensions BridgeRejectObsoleteHeadersAndMessages, RefundBridgeHubKusamaMessages found, treating them as no-effect
    2,764,481: 0xefe5c7aa6fa8283739490110efc6d3cdfcca258d7457a48d3f52b33533985793 (228 bytes) - {"parentHash":"0x3fc6eb0b07c3c7227e816acbdc1b0393400713cecb5c0adfdb4d0506ecca5b7b","number":2764481,"stateRoot":"0xff59bfeccf207980b63cdae0033452b6fdc35b70c7f92b1aa576c855b01019f1","extrinsicsRoot":"0x11544e1f5b0c80a697d52c81e3a04c47b123b1ad6eec38833cd1b0aaa3c88003","digest":{"logs":[{"preRuntime":["0x61757261","0xb796880800000000"]},{"consensus":["0x52505352","0xd17d037540f1e10444fe419b970fb7839d1235b2a4de53e152c6f5823230ae8baa570b05"]},{"seal":["0x61757261","0x8a1974713b7c436351c76f44a47d5e34ff8fd76938fc31b62f6a08c355ce313794b84320edcef3f19a0a5d843380de4eb2841491a8eb05b10e303195e7f59582"]}]}}
    2,764,368: 0xf948499dc823bed3f1434f6907aae9ba835982151f840455e1975c015493618b (228 bytes) - {"parentHash":"0x1ae3761cc8539912d0f2d93f1f8be70762d8b94ad99a45849cb49b9de2a9b646","number":2764368,"stateRoot":"0xc9d7bf39ae63b63fad54efd48c025942897e8fc17c6f96d419af2ae74ec41038","extrinsicsRoot":"0x9c60df17cbb069f7ac31090affdeac0816177e67a99e002c999b8d7d6fdc2989","digest":{"logs":[{"preRuntime":["0x61757261","0x4396880800000000"]},{"consensus":["0x52505352","0xf81a2a37c1349c73e0f449ffd9d97944d4a39860b501f190802df06bfe0fa3021a540b05"]},{"seal":["0x61757261","0x50c57df417af21649b42aa3e38343d5c53692ec046935da415c05283a259783a0119547af255b85dd624665db2831b591a14ffdf0a3d10e1fa0a3b93c1412282"]}]}}
    2,749,318: 0xa4dab582f7413467dcd30c47ffc610bdf682c76b807c29919edff38b71bf1328 (228 bytes) - {"parentHash":"0x2ae76c99cbbd9de6a7956695bce150c49603aa3929638861518c5aa9883f4113","number":2749318,"stateRoot":"0x894f6cf207e6d1104225356ac602ddf91ff201d56484804e2d9ec5c4abc21774","extrinsicsRoot":"0xaf9f362b2bedb863ddc3137d5af473430b03e38b7c3466389aabf311029b3148","digest":{"logs":[{"preRuntime":["0x61757261","0xe659880800000000"]},{"consensus":["0x52505352","0xe526817548a9376ba25e1d51e70a18db1540b0b5c1e9f7e6853028a30f5d616ea6730905"]},{"seal":["0x61757261","0xf2be5c101b502f67e96cded020f7033f7b20f944196ab5ca25ae81719f89be0a5405d2481c2bd2e1696ac80c0b5c3e88a62ee610cfbcef8351fb894b7288338a"]}]}}
    2,729,888: 0x22a5aaf2320c66e10ac68c9b3857225f00d1892bb75439eecb61a3ef5fa8c244 (228 bytes) - {"parentHash":"0x3c7af0f695ee615857654823bd74603d56924008e1f6cfb53f45a07569856ade","number":2729888,"stateRoot":"0xcf35e175b40c13849e915e8d56af44ab0d406846ea9572386fee3a373eafde87","extrinsicsRoot":"0xe89e912abf06a0aef1059029e843f5283d7d9f3c5addd7426486f272604c817d","digest":{"logs":[{"preRuntime":["0x61757261","0xf60b880800000000"]},{"consensus":["0x52505352","0x780759a3465c97bf5b7376f81a154aeaee31d504b4bc2fddcd2835115ac10c45c6080705"]},{"seal":["0x61757261","0xc04db011bc62cd86ca04dddbf44ac812d5754ea39c3a941f3dd87036a5ae5a69f81014f21ff34462b5f2f5a3601f59784832c6701b446a54eaf1e4fd64ef6083"]}]}}
    2,729,833: 0xc9c4dc797655b6763c0cefad438222138beae7f8850ca61253d1427245fcaa86 (228 bytes) - {"parentHash":"0x54c6e05d5a1080040abb03b6535888d42a28e5e5ff314b4e1766e685d1e0a993","number":2729833,"stateRoot":"0x3c8da04093bb5cd21638a89cd34fb5ec91d4adf25d353a0004dbd37ea2ff95dd","extrinsicsRoot":"0x0a2ef1dccbf806c8e2d55036a2f368f91daaee7b2b3a9d6d292eba66d0ad9488","digest":{"logs":[{"preRuntime":["0x61757261","0xbf0b880800000000"]},{"consensus":["0x52505352","0xdfeaa2ec6e3b8eedf9ca0a127df1d7eb74d076c2992558f4d2a05017ec3a439d0e070705"]},{"seal":["0x61757261","0x0428b9e92e291fee45d3046fa5e0b55715d1de96951cf8f2e099e0747fb2cb38aa33d18f69557682945db304144edc48caa03b1720fcec967449d0ee68d91a8f"]}]}}
    2,667,275: 0x7ccebda1ad54e267526ec049bac024295c20925183c7f2db83a87b66a6be4b56 (228 bytes) - {"parentHash":"0x5391e0d9802ecc673953f741ec8352faf766dcc134402063ee48dc7c6683ec89","number":2667275,"stateRoot":"0xdd9e4fcb2e99120c633b655dc07135079f53873295e27ac4ed881ee5f160019b","extrinsicsRoot":"0xfeabfbdbf7aa37babcde3a9d25e02f7cec6a19009c3964ba360d579d2d4492e7","digest":{"logs":[{"preRuntime":["0x61757261","0xf810870800000000"]},{"consensus":["0x52505352","0xe61e1fa61e287fcbc3f6d3f99b0a907b547119287e49404345e7c2b2aa749df9f238ff04"]},{"seal":["0x61757261","0xe4eefb252b6cbcb480c1ca14243e932b5cd753becb848389ea809141d360f5267c541148ea900ffbf06f44587852b697edf62dac6473ddc1c8c4cb253bd40683"]}]}}
    2,627,639: 0x8ece0564d5566401b65f1f9749cd8ac1df6964d819e3a805cea08f8ba7b13c41 (228 bytes) - {"parentHash":"0xe71167e88b79e2fe8f5c958b4128c0efc065ededa3b6f842b4a2df91db6ca04a","number":2627639,"stateRoot":"0x7f09dc977aa091d9ee4514db01ccfc6e943f09749d92f15b9bbf29e76f7c4dde","extrinsicsRoot":"0x84c8e5bc2f942f24afcaf77fae6922f8287ece1891ea9910c0cc69de0572397c","digest":{"logs":[{"preRuntime":["0x61757261","0x0a70860800000000"]},{"consensus":["0x52505352","0x4636685c096953cc8b328d30022ba5615bdf4aa8f38f24737222d23f894c59aa023afa04"]},{"seal":["0x61757261","0x869784136cfccab85062b386c81e47eacdfe7a81f0643eb8a1a113cb6c612f7a4fd8302b94bf5a6b934db7eb34debb339ea29050e19450e6b93882ff298f0d81"]}]}}
    2,627,489: 0xc05b7989d43243f99be5d92d2db64d467bad52c235f7cc86f05806c642f980f6 (228 bytes) - {"parentHash":"0x0c00c1366c28b2980157f9e2fa8715e4980ec755929c2b74212837ad04cf095f","number":2627489,"stateRoot":"0x24476788a9c204381ad33e2771d85e9c515695834056d60441149984240afc5f","extrinsicsRoot":"0xb1098acbc72ce8b2b365e5e41c1520fe4e30eaf661f2c5e3cca9dabfd53ae603","digest":{"logs":[{"preRuntime":["0x61757261","0x716f860800000000"]},{"consensus":["0x52505352","0x762f9d56db7445cadbcebbfc69afc12a8a5b09825d65572ccf1a524ae8e094754235fa04"]},{"seal":["0x61757261","0x349135acc3cc961ce8bdd37973da1a08aa95a11b085e79ac3fad3646f4e3f37945d995f691b00d5e069ba810c4120417592c7f1a3abd22eccb8eef66f0c8d089"]}]}}
    2,623,736: 0x2863a5365aa7543ebba27116af84e4fa56c46e81c8a1e6b127ef668783732c66 (228 bytes) - {"parentHash":"0x867587b3287a459acec68f15f60468ab37e2a4c4d808fa389bd0b7d7636e7b97","number":2623736,"stateRoot":"0x63201f41e23b5155ae4e4d6a381bcfe56a5cbf5c08240d1e3f165b5b13598a04","extrinsicsRoot":"0x6b5e2faa8ec671de3e1cbb99c963a65117933e45896a7a894309396d58cb7e96","digest":{"logs":[{"preRuntime":["0x61757261","0x2e60860800000000"]},{"consensus":["0x52505352","0xd3574711655ec3cf969003aa81fc8eeef426486c8185031581ad561e19d8c65926bcf904"]},{"seal":["0x61757261","0xbc2de34d0db8c75c3716c894bc7ae7cec80714524e1d5f06a4f7a5c8c5e0b00083695392659a34aad5aff343db54d7aa7fd6ed944f739c0e2da762d270292a80"]}]}}
```

It is used for an upcoming feature that allows free header submissions. So, even if some parachain headers exceed 4,096, there is no problem, the relayer will just pay for it instead of not paying.